### PR TITLE
Report CU count instead of WGP for gfx10 winograd RxSf2x3 solver

### PR DESCRIPTION
**Pull request by Slimakanzer**
Resolves issue #654 

This PR provides a function in the Handle to get actual number of _CUs_. This is important for gfx10 GPUs since this generation introduced the concept of _WGP_ (wrokgroup processor), which contain 2 _CUs_. In this case, the OCL and HIP backends report the **number of WGPs instead of CUs**.  Using the number of _WPGs_ instead of _CUs_ in winogradRxSf2x3 gfx10 may have performance regression as mentioned in #654.

## Summary of code changes
* Added `GetMaxHardwareComputeUnits()` to get actual number of CUs instead of WGPs
* Replaced `GetMaxComputeUnits()` with `GetMaxHardwareComputeUnits()` in the winoRxS_f2x3 solver

## Local performance testing
Convolution time improvements for gfx10 compared with #653.  There is no regression with gfx906, ROCm 3.5.1.
* gfx1030, ROCm 3.9
* OCL + HIP backend
* fp16 + fp32

W | H | c | n | k | x | y | p | q | u | v | g | dir | | perf
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
14 | 14 | 1024 | 64 | 1024 | 3 | 3 | 1 | 1 | 1 | 1 | 1 | fwd | | -8%
14 | 14 | 1024 | 64 | 1024 | 3 | 3 | 1 | 1 | 2 | 2 | 1 | fwd | | -8%
14 | 14 | 1024 | 64 | 1024 | 3 | 3 | 1 | 1 | 1 | 1 | 1 | wrw | | 0%
14 | 14 | 1024 | 64 | 1024 | 3 | 3 | 1 | 1 | 1 | 1 | 1 | bwd | | 0%
14 | 14 | 1024 | 64 | 1024 | 3 | 3 | 1 | 1 | 2 | 2 | 1 | bwd | | -8%
56 | 56 | 2048 | 128 | 2048 | 3 | 3 | 1 | 1 | 1 | 1 | 1 | fwd | | 49%
56 | 56 | 2048 | 128 | 2048 | 3 | 3 | 1 | 1 | 2 | 2 | 1 | fwd | | 61%
56 | 56 | 2048 | 128 | 2048 | 3 | 3 | 1 | 1 | 1 | 1 | 1 | bwd | | 49%
56 | 56 | 2048 | 128 | 2048 | 3 | 3 | 1 | 1 | 2 | 2 | 1 | bwd | | 50%
56 | 56 | 2048 | 128 | 2048 | 3 | 3 | 1 | 1 | 1 | 1 | 1 | wrw | | 0%
224 | 224 | 512 | 64 | 512 | 3 | 3 | 1 | 1 | 1 | 1 | 1 | fwd | | 47%
224 | 224 | 512 | 64 | 512 | 3 | 3 | 1 | 1 | 2 | 2 | 1 | fwd | | 47%
224 | 224 | 512 | 64 | 512 | 3 | 3 | 1 | 1 | 1 | 1 | 1 | bwd | | 47%
224 | 224 | 512 | 64 | 512 | 3 | 3 | 1 | 1 | 2 | 2 | 1 | bwd | | 47%
224 | 224 | 512 | 64 | 512 | 3 | 3 | 1 | 1 | 1 | 1 | 1 | wrw | | 32%

### CODE REVIEW

File: src/solver/conv_winoRxS_f2x3.cpp [hunk 1]
```diff
@@ -176,7 +176,7 @@ inline bool IsShaderContraintsMet(const int R,
         if(!(0 <= params.GetBackwardPadH() && params.GetBackwardPadH() < std::pow(2, 16)))
             return false;
     }
-    const auto grid_workgroup_count_x = params.GetStream().GetMaxComputeUnits();
+    const auto grid_workgroup_count_x = params.GetStream().GetMaxHardwareComputeUnits();
```
Comment (atamazov) [inline src/solver/conv_winoRxS_f2x3.cpp:179]: Is it so that HW CUs is required only for this Solver or we shall expect more changes in the future?

**Comment by atamazov** [comment]
Please resolve merge conflicts

File: src/solver/conv_winoRxS_f2x3.cpp (hunk 1 above)
Comment (Slimakanzer) [reply src/solver/conv_winoRxS_f2x3.cpp:179]: In other solvers, either the gfx10 is not supported, or HW CU is not. So HW CUs not used anywhere else at the moment.

**Comment by atamazov** [comment]
@Kirpich30000 Please review.

**Review by Kirpich30000 — APPROVED** [review]
LGTM

**Pull request merged**